src/modules/control/src/waypoint_rviz_visualizer.py

```diff
@@ -0,0 +1,117 @@
+
+        msg_scale = rospy.get_param('/virtual_donut_node/msg_scale')
+        
+        inner_donut_r = rospy.get_param('/virtual_donut_node/inner_donut_r')
+        inner_donut_g = rospy.get_param('/virtual_donut_node/inner_donut_g')
+        inner_donut_b = rospy.get_param('/virtual_donut_node/inner_donut_b')
+        inner_donut_a = rospy.get_param('/virtual_donut_node/inner_donut_a')
+
+        outer_donut_r = rospy.get_param('/virtual_donut_node/outer_donut_r')
+        outer_donut_g = rospy.get_param('/virtual_donut_node/outer_donut_g')
+        outer_donut_b = rospy.get_param('/virtual_donut_node/outer_donut_b')
+        outer_donut_a = rospy.get_param('/virtual_donut_node/outer_donut_a')
+
+        self.innerDonutMsg.scale.x = msg_scale
+        self.innerDonutMsg.scale.y = msg_scale
+        self.innerDonutMsg.scale.z = msg_scale
+
+        self.outerDonutMsg.scale.x = msg_scale
+        self.outerDonutMsg.scale.y = msg_scale
+        self.outerDonutMsg.scale.z = msg_scale
+
+        self.innerDonutMsg.type = 8
+        self.outerDonutMsg.type = 8
+
+        self.innerDonutMsg.color.a = inner_donut_a
+        self.innerDonutMsg.color.r = inner_donut_r
+        self.innerDonutMsg.color.g = inner_donut_g
+        self.innerDonutMsg.color.b = inner_donut_b
+        
+        self.outerDonutMsg.color.a = outer_donut_a
+        self.outerDonutMsg.color.r = outer_donut_r
+        self.outerDonutMsg.color.g = outer_donut_g
+        self.outerDonutMsg.color.b = outer_donut_b
+
+        if self.point_initialization_state:
+            # calculate the placement of each point in polar coordinates
+            for i in range(0, point_density):
+                point_density_increment = 360 / point_density
+
+                x_inner_donut = inner_donut_bound * math.cos(point_density_increment*i)
+                y_inner_donut = inner_donut_bound * math.sin(point_density_increment*i)
+
+                x_outer_donut = outer_donut_bound * math.cos(point_density_increment*i)
+                y_outer_donut = outer_donut_bound * math.sin(point_density_increment*i)
+
+                self.innerDonutMsg.points.append(Point(x_inner_donut, y_inner_donut, 0))
+                self.outerDonutMsg.points.append(Point(x_outer_donut, y_outer_donut, 0))
+            
+            self.point_initialization_state = False
+
+        self.innerDonutMsg.header.frame_id = "base_link"
+        self.outerDonutMsg.header.frame_id = "base_link"
+
+    def create_virtual_donut(self):
+        """Publish marker messages in the base_link frame"""
+        self.innerDonutMsg.header.stamp = rospy.Time.now()
+        self.outerDonutMsg.header.stamp = rospy.Time.now()
+
+        self.innerDonutPublisher.publish(self.innerDonutMsg)
+        self.outerDonutPublisher.publish(self.outerDonutMsg)
+        
+    def virtual_donut_creation(self):
+        """Create marker messages and publish to rviz for visualization"""
+        rospy.Timer(rospy.Duration(0.025), self.vdonut_callback)
```

Comment:
Maybe use `rospy.Rate()` instead, or save a variable for Hz, since it seems like that is variable you actually care able [(line 28)](https://github.com/SJSU-AD/FusionAD/blob/a697b8ec01862d9d1b5e26a6a5402de2b0505daa/src/modules/control/src/waypoint_rviz_visualizer.py#L28)

Reply:
Also the rate of specifically 40 Hz in the docstring in [line 28](https://github.com/SJSU-AD/FusionAD/blob/a697b8ec01862d9d1b5e26a6a5402de2b0505daa/src/modules/control/src/waypoint_rviz_visualizer.py#L28) can get easily outdated. Maybe leave that number out of the docstring and note it as a variable elsewhere, or a comment much closer to this line, where the value actually affects the code.

Reply:
I can change the duration to a configurable parameter in the launch file and remove the comment

Reply:
Cool even better

Reply:
The [arg tag](http://wiki.ros.org/roslaunch/XML/arg) might come in handy for this